kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/InsecureTls.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault.config;
+
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * A {@link TrustProvider} that allows the disabling trust verification.
+ * <strong>Not recommended for production use.</strong>
```

Comment:
I think we should have a stronger warning. 
```suggestion
 * <strong>Should not be used outside of automated tests.</strong>
```

Is there any way we can shift if out of the production source tree?

Reply:
I wouldn't shift InsecureTls out from the tree.   

- it is feature people expect to find amongst TLS configuration options. 
- we've done our bit and documented the recommendation not to use in production.  I think the responsibility now rests with the production deployer to make good choices.
- i know I've valued have insecureTls options to aid problem diagnosis. it is a useful blade to have on the swiss army knife.

Reply:
If we don't supply a secure version by default though I worry people will just run this anyway. Sonar threw its toys out of the pram about it.

I'd be quite comfortable if we had it as an optional module that required something additional to enable it. I just worry a lot about providing a proxy which messes trust/tls up in production

Reply:
+1 to having it available and documented as dangerous.

What would the secure version by default be for a Vault instance with private certs? They are going to have to supply a trust store and configure something.

Reply:
I've refactored it again to share the configuration classes that we use for configuring upstream/downstream TLS in the kroxy runtime. So for this PR I think they should remain consistent.